lib/github-pages-health-check.rb

```diff
@@ -48,6 +48,31 @@ module HealthCheck
       }
     }.freeze
 
+    class << self
+      def default_resolver
+        Dnsruby::Resolver.new(
+          retry_times: 2,
+          query_timeout: 2,
+          dnssec: true,
+        )
+      end
+
+      def nameservers_for_domain(domain)
+        default_resolver.query(domain, Dnsruby::Types::NS).answer.map do |answer|
+          next answer.nsdname if answer.type == Dnsruby::Types::NS
+        end.compact
+      end
+
+      def build_resolver(domain)
+        Dnsruby::Resolver.new(
+          nameservers: nameservers_for_domain(domain),
+          retry_times: 2,
+          query_timeout: 2,
+          dnssec: true,
```

Comment:
To DRY things up, perhaps move the default options to a class-level constant/hash and just merge the nameservers in here?